src/Uno.UI.Composition/Composition/CompositionEffectBrush.skia.cs

```diff
@@ -0,0 +1,1726 @@
+﻿#nullable enable
+
+using System;
+using SkiaSharp;
+using Windows.UI;
+using Microsoft.UI;
+using System.Numerics;
+using Uno.UI.Composition;
+using Windows.Graphics.Effects;
+using Windows.Graphics.Effects.Interop;
+using System.Diagnostics;
+
+namespace Windows.UI.Composition;
+
+public partial class CompositionEffectBrush : CompositionBrush
+{
+	private bool _isCurrentInputBackdrop;
+	private bool? _currentCompMode;
+
+	private SKRect _currentBounds;
+	private SKImageFilter? _filter;
+
+	internal bool HasBackdropBrushInput { get; private set; }
+
+	internal bool UseBlurPadding { get; set; }
+
+	private SKImageFilter? GenerateEffectFilter(object? effect, SKRect bounds, bool tryUsingSourceSize = false)
+	{
+		// TODO: https://user-images.githubusercontent.com/34550324/264485558-d7ee5062-b0e0-4f6e-a8c7-0620ec561d3d.png
+		// TODO: Cache pixel shaders (see dwmcore.dll!CCompiledEffectCache), needed in order to implement animations and online rendering
+
+		switch (effect)
+		{
+			case CompositionEffectSourceParameter effectSourceParameter:
+				{
+					CompositionBrush? brush = GetSourceParameter(effectSourceParameter.Name);
+					if (brush is not null)
+					{
+						if (brush is CompositionBackdropBrush)
+						{
+							_isCurrentInputBackdrop = true;
+							HasBackdropBrushInput = true;
+							return null;
+						}
+
+						_isCurrentInputBackdrop = false;
+
+						SKRect srcBounds = bounds;
+						if (tryUsingSourceSize && brush is ISizedBrush { IsSized: true, Size: Vector2 size })
+						{
+							srcBounds = bounds with { Right = size.X, Bottom = size.Y };
+						}
+
+						SKPaint paint = new SKPaint() { IsAntialias = true, IsAutohinted = true, FilterQuality = SKFilterQuality.High };
+						brush.UpdatePaint(paint, srcBounds);
+
+						return SKImageFilter.CreatePaint(paint, new(srcBounds));
+					}
+
+					return null;
+				}
+			case IGraphicsEffectD2D1Interop effectInterop:
+				{
+					switch (EffectHelpers.GetEffectType(effectInterop.GetEffectId()))
+					{
+						case EffectType.GaussianBlurEffect:
+							{
+								if (effectInterop.GetSourceCount() == 1 && effectInterop.GetPropertyCount() == 3 && effectInterop.GetSource(0) is IGraphicsEffectSource source)
+								{
+									SKImageFilter? sourceFilter = GenerateEffectFilter(source, bounds);
+									if (sourceFilter is null && !_isCurrentInputBackdrop)
+									{
+										return null;
+									}
+
+									_isCurrentInputBackdrop = false;
+
+									effectInterop.GetNamedPropertyMapping("BlurAmount", out uint sigmaProp, out _);
+									effectInterop.GetNamedPropertyMapping("Optimization", out uint optProp, out _);
+									effectInterop.GetNamedPropertyMapping("BorderMode", out uint borderProp, out _);
+
+									float sigma = (float)(effectInterop.GetProperty(sigmaProp) ?? throw new NullReferenceException("The effect property was null"));
```

Comment:
For all this PR, let's not throw `NullReferenceException`, use `InvalidOperationException` instead.